core/blockchain.go

```diff
@@ -1854,50 +1864,70 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 		statedb.StartPrefetcher("chain")
 		activeState = statedb
 
-		// If we have a followup block, run that against the current state to pre-cache
-		// transactions and probabilistically some of the account/storage trie nodes.
-		var followupInterrupt uint32
-		if !bc.cacheConfig.TrieCleanNoPrefetch {
-			if followup, err := it.peek(); followup != nil && err == nil {
-				throwaway, _ := state.New(parent.Root, bc.stateCache, bc.snaps)
-
-				go func(start time.Time, followup *types.Block, throwaway *state.StateDB, interrupt *uint32) {
-					bc.prefetcher.Prefetch(followup, throwaway, bc.vmConfig, &followupInterrupt)
-
-					blockPrefetchExecuteTimer.Update(time.Since(start))
-					if atomic.LoadUint32(interrupt) == 1 {
-						blockPrefetchInterruptMeter.Mark(1)
+		accounts := make(map[common.Address]bool, block.Transactions().Len())
+		accountsSlice := make([]common.Address, 0, block.Transactions().Len())
+		for _, tx := range block.Transactions() {
+			from, err := types.Sender(signer, tx)
+			if err != nil {
+				break
+			}
+			accounts[from] = true
+			if tx.To() != nil {
+				accounts[*tx.To()] = true
+			}
+		}
+		for account, _ := range accounts {
+			accountsSlice = append(accountsSlice, account)
+		}
+		if len(accountsSlice) >= preLoadLimit {
+			objsChan := make(chan []*state.StateObject, runtime.NumCPU())
+			for i := 0; i < runtime.NumCPU(); i++ {
```

Comment:
should we make sure that `len(accountsSlice)` > `runtime.NumCPU()`

Reply:
Add one more condition: len(accountsSlice) > runtime.NumCPU()

Reply:
Why did we not add the conditional here? Forgot maybe?